admin/acceptance.py

```diff
@@ -1212,25 +1227,27 @@ def main(reactor, args, base_path, top_level):
         result = 1
         raise
     finally:
-        # Unless the tests failed, and the user asked to keep the nodes, we
-        # delete them.
-        if not options['keep']:
-            runner.stop_cluster(reactor)
+        reached_finally = True
+        # We delete the nodes if the user hasn't asked to keep them
+        # or if we failed to provision the cluster.
+        if not setup_succeeded:
+            print "cluster provisioning failed"
+        elif not options['keep']:
+            print "not keeping cluster"
         else:
             print "--keep specified, not destroying nodes."
-            if cluster is None:
-                print ("Didn't finish creating the cluster.")
-            else:
-                print ("To run acceptance tests against these nodes, "
-                       "set the following environment variables: ")
+            assert cluster is not None
```

Comment:
This should be an `if` with some well-defined failure behavior.  `assert` might get "optimized" away if Python is run in a certain way and, in general, we just don't use `assert` anywhere in Flocker.